pynetworktables2js/js/tableviewer.js

```diff
@@ -0,0 +1,290 @@
+"use strict";
+
+(function() {
+
+	if ($ === undefined) {
+	    alert("jQuery must be downloaded+included to use tableviewer.js!");
+	    return;
+	}
+
+	function Tableviewer($el) {
+		this.$el = $el.addClass('tableviewer');
+		this.ntRoot = {
+			'' : {
+				type : 'table',
+				$el : $('<ul>Root</ul>').appendTo($el)
+			}
+		};
+
+		// Expand/Collabse tables
+		$el.on('click', '.expanded, .collapsed', function(e) {
+			$(this).toggleClass('expanded collapsed');
+		});
+
+		// Resize number input based on length of input
+		$el.on('change keyup ntUpdate', '[type=number]', function(e) {
+			var text = $(this).val();
+			var length = text.length;
+			var $phantomInput = $(this).next();
+			$phantomInput.text(text);
+			$(this).css('max-width', 13 + $phantomInput.width());
+		});
+
+		$el.on('change keyup ntUpdate', '[type=text]', function(e) {
+			var text = $(this).val();
+			var length = text.length;
+			var $phantomInput = $(this).next();
+			$phantomInput.text(text);
+			//$(this).attr('size', length);
+			$(this).css('max-width', 10 + $phantomInput.width());
+		});
+
+		// Update NetworkTables
+		$el.on('change', '[type=checkbox]', function(e) {
+			var key = $(this).parents('[data-path]').data('path');
+			var value = $(this).prop('checked');
+			NetworkTables.putValue(key, value);
+		});
+
+		$el.on('change', '[type=text]', function(e) {
+			var key = $(this).parents('[data-path]').data('path');
+			var value = $(this).val();
+			NetworkTables.putValue(key, value);
+		});
+
+		$el.on('change', '[type=number]', function(e) {
+			var key = $(this).parents('[data-path]').data('path');
+			var value = parseFloat($(this).val());
```

Comment:
The input is a number type, so I'm not sure why it would fail. But if parseFloat is given something it can't parse then value will be NaN. That will raise a value error on the server.